tensorrt_llm/_torch/auto_deploy/transformations/transform.py

```diff
@@ -135,14 +145,48 @@ def __call__(self, cm: CachedSequenceInterface) -> GraphModule:
             except ImportError:
                 pass
 
+        ############################################################################################
+        # RESIZE CACHE
+        ############################################################################################
+        free_mem, total_mem = torch.cuda.mem_get_info()
+        ad_logger.info(f"Free memory: {free_mem}, Total memory: {total_mem}")
+        current_cache_size = cm.current_cache_size_bytes()
+        current_num_pages = cm.info.num_pages
+        ad_logger.info(
+            f"Current cache size: {current_cache_size}, Current num pages: {current_num_pages}"
+        )
+
+        # Let's run a forward pass to get the memory usage
+        cm.info._set_max_num_tokens_sample()
+        free_mem_pre, _ = torch.cuda.mem_get_info()
+        ad_logger.info(f"Free memory before forward pass: {free_mem_pre}")
+        egm(*cm.args)
+        free_mem_post, _ = torch.cuda.mem_get_info()
+        ad_logger.info(f"Free memory after forward pass: {free_mem_post}")
+        memory_for_forward_pass = free_mem_pre - free_mem_post
+        ad_logger.info(f"Memory for forward pass: {memory_for_forward_pass}")
+
+        # FIXME: 0.8 is hard coded to ensure we have enough memory for graph capture.
+        new_cache_size = free_mem_post * 0.8 + current_cache_size
+        new_num_pages = int(new_cache_size // (current_cache_size // current_num_pages))
+        ad_logger.info(f"New cache size: {new_cache_size}, New num pages: {new_num_pages}")
+        cm.resize_cache(new_num_pages)
+
         ############################################################################################
         # COMPILE MODEL
         ############################################################################################
 
         cm.info._set_generate_only_batch()
+        compiler_kwargs = {"cuda_graph_batch_sizes": self.ad_config.cuda_graph_batch_sizes}
         egm_compiled = compile_and_capture(
-            egm, self.compile_backend, args=cm.args, dynamic_shapes=cm.dynamic_shapes
+            egm,
+            self.compile_backend,
+            args=cm.args,
+            kwargs=compiler_kwargs,
+            dynamic_shapes=cm.dynamic_shapes,
```

Comment:
`kwargs` is used for model kwargs in this context. how about just this signature instead:
```suggestion
            egm,
            self.compile_backend,
            args=cm.args,
            dynamic_shapes=cm.dynamic_shapes,
            cuda_graph_batch_sizes=self.ad_config.cuda_graph_batch_sizes
```

Reply:
You can update the `BackendCompiler` `__init__` signature like this to accommodate for it:

```
class BackendCompiler(ABC):
    max_batch_size: int
    def __init__(
        self,
        gm: GraphModule,
        args: Tuple[Any, ...],
        kwargs: Optional[Dict[str, Any]] = None,
        dynamic_shapes=None,
        **compiler_kwargs
    ):
```


And then the `TorchOptCompiler` class can have the following `__init__`:


```
@BackendRegistry.register("torch-opt")
class TorchOptCompiler(BackendCompiler):
    def __init__(self, *args, cuda_graph_batch_sizes, **kwargs):
        super().__init__(*args, **kwargs)
        self.cuda_graph_batch_sizes = cuda_graph_batch_sizes
```

Reply:
done. ptal